pkg/server/plugin/datastore/sql/migration.go

```diff
@@ -27,6 +40,23 @@ func migrateDB(db *gorm.DB, dbType string, log hclog.Logger) (err error) {
 		return initDB(db, dbType, log)
 	}
 
+	// TODO related epic https://github.com/spiffe/spire/issues/1083
+	// We do not allow breaking changes in 0.X beyond version 0.9.0, but
+	// 1.0+ may have such changes. Thus iff Major version is 0, we are
+	// compatible. This logic will change in 1.0+.
+	// With 1.0 code, we will need a proper version check between code and db versions.
+	if codeVersion.Major != 0 {
+		err = sqlError.New("current migration code not compatible with %s release", codeVersion.String())
+		log.Error(err.Error())
+		return err
+	}
+
+	// TODO related epic https://github.com/spiffe/spire/issues/1083
+	// continue doing this automigration for now
+	// eventually SPIRE Code will be bumped to version 0.10, where we can
+	// expect that the `migrations` table exists and has the CodeVersion column,
+	// and then this can be removed. The below `Assign` check may be a new guard
+	// against pre-0.9 versions due to mismatching model
```

Comment:
I think we need to keep this here even post-0.10 because we will need it when starting a fresh cluster?

Reply:
true true